--- a/ext/libav/gstavviddec.c
+++ b/ext/libav/gstavviddec.c
@@ -2064,6 +2064,13 @@
     if (strstr (in_plugin->name, "vaapi")) {
       GST_DEBUG
           ("Ignoring VAAPI decoder %s. We can't handle this outside of ffmpeg",
+          in_plugin->name);
+      goto next;
+    }
+
+    if (g_str_has_suffix (in_plugin->name, "_qsv")) {
+      GST_DEBUG
+          ("Ignoring qsv decoder %s. We can't handle this outside of ffmpeg",
           in_plugin->name);
       goto next;
     }
--- a/ext/libav/gstavvidenc.c
+++ b/ext/libav/gstavvidenc.c
@@ -974,12 +974,19 @@ gst_ffmpegvidenc_register (GstPlugin * p
       goto next;
     }
 
-    if (g_str_has_suffix (in_plugin->name, "_nvenc")) {
+    if (strstr (in_plugin->name, "nvenc")) {
       GST_DEBUG
           ("Ignoring nvenc encoder %s. We can't handle this outside of ffmpeg",
           in_plugin->name);
       goto next;
     }
+
+    if (g_str_has_suffix (in_plugin->name, "_qsv")) {
+      GST_DEBUG
+          ("Ignoring qsv encoder %s. We can't handle this outside of ffmpeg",
+          in_plugin->name);
+      goto next;
+    }
 
     /* only video encoders */
     if (!av_codec_is_encoder (in_plugin)
