diff -up gst-libav-1.1.3/ext/libav/gstavmux.c~ gst-libav-1.1.3/ext/libav/gstavmux.c
--- gst-libav-1.1.3/ext/libav/gstavmux.c~	2013-07-15 08:50:20.000000000 +0200
+++ gst-libav-1.1.3/ext/libav/gstavmux.c	2013-08-08 22:00:08.311276449 +0200
@@ -939,6 +939,8 @@ gst_ffmpegmux_register (GstPlugin * plug
     while (*p) {
       if (*p == '.')
         *p = '_';
+      if (*p == ',')
+        *p = '_';
       p++;
     }
 
